src/python/pants/backend/jvm/tasks/unpack_jars.py

```diff
@@ -57,6 +57,12 @@ class MissingUnpackedDirsError(Exception):
   def product_types(cls):
     return ['unpacked_archives']
 
+  @classmethod
+  def register_options(cls, register):
+    super(UnpackJars, cls).register_options(register)
+    register('--with-dependencies', type=bool, default=True,
```

Comment:
... maybe the idea is this is advanced, and so you can install the task in a new goal that also includes the consumer (to the right in the goal list of tasks), and you could then use pants.ini to lock that installation of the task into intransitive?

Reply:
That Android code actually significantly overrode the UnpackJars logic, IIRC, because it didn't map well to the Android paradigm. UnpackJars unpacks and filters for every dependency (again, IIRC. I am AFK so going largely off memory), and didn't cache the unzipped result. The Android code unpacked to a workdir and wouldn't unpack again until the filter changed or the version changed.